download-results.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python3
+
+# Script for downloading results from the OpenScPCA-nf workflow results bucket
+
+import argparse
+import datetime
+import os
+import pathlib
+import re
+import subprocess
+import sys
+from typing import List, Set
+
+# import functions from download-data.py
+download_data = __import__("download-data")
+
+# enable text formatting on Windows
+os.system("")
+
+RESULTS_BUCKET = "openscpca-nf-workflow-results"
+TEST_RESULTS_BUCKET = "openscpca-test-workflow-results-public-access"
+
+
+def get_results_modules(bucket: str, release: str, profile: str) -> List[str]:
+    """
+    Get the list of available results modules from the OpenScPCA workflow results bucket.
+    """
+    ls_cmd = ["aws", "s3", "ls", f"s3://{bucket}/{release}/"]
+    if profile:
+        ls_cmd += ["--profile", profile]
+    if bucket == TEST_RESULTS_BUCKET:
+        ls_cmd += ["--no-sign-request"]
```

Comment:
Reporting a result I was not expecting (have not looked into it further yet):
```
% ./download-results.py --test-data

Error listing release versions from the OpenScPCA bucket.

Make sure you have the correct profile active (or use the --profile option) and run `aws sso login` before running this script.

AWS Error:  
Unable to locate credentials. You can configure credentials by running "aws configure".
```

Reply:
When I run this, I get the following which makes me think it might have something to do with the ability to access the test bucket? 

```
./download-results.py --test-data --dryrun --profile openscpca
Error listing release versions from the OpenScPCA bucket.

Make sure you have the correct profile active (or use the --profile option) and run `aws sso login` before running this script.

AWS Error:  
An error occurred (AccessDenied) when calling the ListObjectsV2 operation: Access Denied
```

Reply:
This works just fine: `aws s3 ls s3://openscpca-test-workflow-results-public-access/test --no-sign-request`

`aws s3 ls s3://openscpca-test-workflow-results-public-access --profile openscpca` returns: 

```
An error occurred (AccessDenied) when calling the ListObjectsV2 operation: Access Denied
```

Which is what I would expect.

Reply:
Confirming that this now works as expected with the updates.